### ISSUE TITLE
Quick Share support?

### ISSUE THREAD

**Issue by Gregman-js** [grishka/NearDrop#140]
Hello 
Does NearDrop work with the new nearby share implementation, aka Quick Share?
My experiments show that no :(
For the uninitiated, Google is joining forces with Samsung and introducing a joint alternative in the form of Quick Share. It's rolling out now to supported devices.

My test was trying to share a photo from the Samsung S23 to the MacBook with neardrop v 2.0.2
It works with nearby share, but choosing new unified Quick Share — it doesn't detect MacBook
I'm curious how this new application works from the inside and whether it will be possible to implement these protocols here

**Comment by ubdjshdb** [comment]
>It works with nearby share, but choosing new unified Quick Share — it doesn't detect MacBook

From my testing, Nearby Share devices only appear in Samsung Quick Share only after Nearby Share is removed (after a GMS update or a remote config flag?). Tested on S21+/S23 with both NearDrop on Mac and the official app on Windows: they were only showing up in Nearby Share yesterday. Got the update that removed Nearby Share today, now they're showing up in the Samsung app.
For non-Samsung devices, I'm pretty sure it's just a rebranding.

Also, this update breaks receiving files from NearDrop on Samsung devices since the Receive files screen is gone (not sure about non-Samsung devices).

**Comment by grishka** [comment]
> update that removed Nearby Share

That's a thing now?!

**Comment by ubdjshdb** [comment]
> > update that removed Nearby Share
> 
> That's a thing now?!

On Samsung devices, yep. On other phones it rebrands Nearby Share to Quick Share.

**Comment by julianebert** [comment]
+1 After updating Nearby Share to Quick Share on my Pixel 8 Pro, my Macbook Pro with Neardrop doesn't show up in the phone's share menu.

**Comment by GalacticSenat** [comment]
IT WORKS

**Comment by jonathandhn** [comment]
It doesn't work with samsung galaxy s23 since the merge, I did not have access to google's nearby share any more only Quick Share

**Comment by albertonoys** [comment]
Same here thing here on a Galaxy S23.

**Comment by GalacticSenat** [comment]
Im on s21 Ultra and it works for me, are you guys on newest quickshare version?

**Comment by albertonoys** [comment]
> Im on s21 Ultra and it works for me, are you guys on newest quickshare version?

Yeah, it says I'm on the latest versions:

Quick share: 13.6.11.7
Quick share connectivity: 1.5.02.30
Quick share agent: 3.5.19.23

**Comment by grishka** [comment]
> It doesn't completely remove Nearby Share, it merges Nearby Share with Samsung's own file sharing tool.

The only thing that matters in relation to NearDrop is whether the network protocol is different. Or, at least backwards compatible enough (those version numbers in all packets are there for a reason, after all).

And no, I'm not renaming it to QuickDrop :P

**Comment by jonathandhn** [comment]
hi @grishka,

So using the QuickShare option to no avail, even with other device (non samsung) using google nearby share. When using another android phone and enabling nearby share discovery, it triggers the google's services pop-up to open the old nearby share. 

Using the old nearby share on the samsung s23 (without using the direct share button that has disappeared) allow you to see your mac, but it fails to show a pop-up on the macos side. Moreover, it ended up with multiple time showing duplicated computer (for a same and single mac, with the same and single near drop launched). 

Both Xiaomi android 12 and Samsung S23 android 14 now fail to connect to neardrop app, they have changed something on both side, and google nearby share is not actually compatible with Samsung quickshare (even if in the settings of the latter, there is written your samsung account for samsung, your google account for google sharing) as it is not showing android devices on the quickshare menu, and other device sharing with a quickshare enabled device default to google's services. What a mess they have done, it basically means everyone keeps its own tech, but under a single name and a single logo.

**Comment by grishka** [comment]
> but it fails to show a pop-up on the macos side

NearDrop doesn't implement the "device nearby is sharing" popup. Although I suppose I could add it? I reverse engineered that part. It relies on BLE. I can't **send** the required broadcasts from macOS because of its limited APIs, but there's nothing to prevent me from **receiving** them.

> and google nearby share is not actually compatible with Samsung quickshare

You had one fucking job, Google...

**Comment by jonathandhn** [comment]
@grishka 
> > but it fails to show a pop-up on the macos side
> 
> NearDrop doesn't implement the "device nearby is sharing" popup. Although I suppose I could add it? I reverse engineered that part. It relies on BLE. I can't **send** the required broadcasts from macOS because of its limited APIs, but there's nothing to prevent me from **receiving** them.
> 
I mean, when I choose a file to share with my computer, the pop-up to accept the transfer is not showing any more on macOS, nor the device actually findable from QuickShare. 

Actually the device is findable from the old Nearby Share Menu on Samsung, And Nearby Share menu on the Xiaomi, but the pop-up to accept the transfer is not showing any more on macOS.

**Comment by grishka** [comment]
Does it just show up as loading and then fails with a timeout?

**Comment by jonathandhn** [comment]
On macOS side, nothing, and on android side timeout

https://github.com/grishka/NearDrop/assets/55917666/875709a4-fd81-49b3-a860-c3c27fb289de

https://github.com/grishka/NearDrop/assets/55917666/ea17dd07-760b-4455-bcb3-102f81665cc0

**Comment by dohun0310** [comment]
On the Galaxy Z Flip3 I'm using, it doesn't work with 'Quick Share', but it does work with 'Nearby Share'. As you can see in the video below, if I try it on 'Quick Share' it says 'Fail' and I don't even get any notification on my MacBook Pro (Sonoma 14.3), whereas on 'Nearby Share' it works very well. I also moved all the videos below via 'NearDrop' and 'Nearby Share'.

** Note ** If you're not sure if 'Quick Share' and 'Nearby Share' are integrated on your Samsung Galaxy device, check to see if the icons are the same as mine, or long press on 'Quick Share' in the quick panel and a modal will come up telling you they are integrated.

Quick Share Verison: 13.6.11.7
Quick Share Agent Version: 3.5.9.23
Quick Share Connectivity Version: 1.5.02.30
Google Play System Version: November 1, 2023

https://github.com/grishka/NearDrop/assets/63141348/d27dd38f-f215-47cf-bca7-84e8ffe0507c


https://github.com/grishka/NearDrop/assets/63141348/de74798a-f51d-44fc-b247-3068add34195

**Comment by jonathandhn** [comment]
@dohun0310

> On the Galaxy Z Flip3 I'm using, it doesn't work with 'Quick Share', but it does work with 'Nearby Share'. As you can see in the video below, if I try it on 'Quick Share' it says 'Fail' and I don't even get any notification on my MacBook Pro (Sonoma 14.3), whereas on 'Nearby Share' it works very well. I also moved all the videos below via 'NearDrop' and 'Nearby Share'.
> 
> ** Note ** If you're not sure if 'Quick Share' and 'Nearby Share' are integrated on your Samsung Galaxy device, check to see if the icons are the same as mine, or long press on 'Quick Share' in the quick panel and a modal will come up telling you they are integrated.
> 
> Quick Share Verison: 13.6.11.7 Quick Share Agent Version: 3.5.9.23 Quick Share Connectivity Version: 1.5.02.30 Google Play System Version: November 1, 2023
> 
>  Screen_Recording_20240205_224035_Google.Play.services.mp4 
> 
>  Screen_Recording_20240205_223540_Quick.Share.mp4

But you can see your mac on the device list if QuicShare ? Because mine is not to be found. 

As of today @grishka something happened, nearby share seams to work back again with S23, there is something strange going server side I think.

**Comment by dohun0310** [comment]
> @dohun0310
> 
> > On the Galaxy Z Flip3 I'm using, it doesn't work with 'Quick Share', but it does work with 'Nearby Share'. As you can see in the video below, if I try it on 'Quick Share' it says 'Fail' and I don't even get any notification on my MacBook Pro (Sonoma 14.3), whereas on 'Nearby Share' it works very well. I also moved all the videos below via 'NearDrop' and 'Nearby Share'.
> > ** Note ** If you're not sure if 'Quick Share' and 'Nearby Share' are integrated on your Samsung Galaxy device, check to see if the icons are the same as mine, or long press on 'Quick Share' in the quick panel and a modal will come up telling you they are integrated.
> > Quick Share Verison: 13.6.11.7 Quick Share Agent Version: 3.5.9.23 Quick Share Connectivity Version: 1.5.02.30 Google Play System Version: November 1, 2023
> > Screen_Recording_20240205_224035_Google.Play.services.mp4
> > Screen_Recording_20240205_223540_Quick.Share.mp4
> 
> But you can see your mac on the device list if QuicShare ? Because mine is not to be found.
> 
> As of today @grishka something happened, nearby share seams to work back again with S23, there is something strange going server side I think.

Yes, I can see my MacBook even if I use Quick Share. I want to see what happens using XCode, but it's a bit difficult because I don't know how to develop macOS and iOS.

**Comment by dohun0310** [comment]
@grishka
I am unable to send files via 'NearDrop' anymore because 'Nearby Share' has been completely removed and merged into 'Quick Share' with the installation of One UI 6.1 on Galaxy devices. Upon checking my Galaxy S24 Ultra (running One UI 6.1) using the Files app, I received the following toast message.

'Nearby Share is not available on this device.'

![Screenshot_20240206_221528_Files by Google](https://github.com/grishka/NearDrop/assets/63141348/a0196f8f-77ce-4577-b84f-8143aad28f69)

**Comment by kineticscreen** [comment]
Unfortunately Google has also removed the "Receive" page from the Files app, so you can't use it to make your phone visible for receiving from NearDrop

**Comment by NayamAmarshe** [comment]
Google dropped the Receive thing from the Files app. What do we do now? :/

**Comment by grishka** [comment]
> Google dropped the Receive thing from the Files app. What do we do now? :/

Oops. Make in intent shortcut like it says in readme then. Unless they removed that activity as well.

**Comment by dohun0310** [comment]
> > Google dropped the Receive thing from the Files app. What do we do now? :/
> 
> Oops. Make in intent shortcut like it says in readme then. Unless they removed that activity as well.

They started to remove that activity in gms

**Comment by grishka** [comment]
That's extremely unfortunate. But then maybe they added some new way to enable the MDNS service that doesn't involve sending impossible BLE advertisements. This needs further research.

**Comment by mitesh77** [comment]
I am unable to send files via 'NearDrop' anymore because 'Nearby Share' has been completely removed and merged into 'Quick Share' with using google Files app, I am using google Pixel 8 pro now not working at all we don't have any Receive Button so how we fixed?

**Comment by mitesh77** [comment]
@grishka I think at the moment the best solution is to keep the old Google Files app apk, installation takes a few seconds and then open and use It. 

another solution is we can just clean the storage and cache whenever mac to sends Android file share.

https://github.com/grishka/NearDrop/assets/38649435/238f94b9-b61c-45e1-8971-419438d2356a


https://github.com/grishka/NearDrop/assets/38649435/40642320-2ced-4781-8984-6d3b7200de3b

it is not appropriate for the long term but is Ok for now.

**Comment by stanleylhs** [comment]
@mitesh77 Which version should we install to get the receive tab back?

**Comment by souljedi** [comment]
Maybe the following works for some of you as well (as a workaround):

In the Google Play Store, navigate to Google Files and then select "Uninstall". This will not remove Google Files but it will uninstall all updates to it. Afterwards, I got the Nearby receive tab again. Furthermore, you can disable automatic updates for that package, until hopefully a new solution is found.

**Comment by mitesh77** [comment]
> @mitesh77 Which version should we install to get the receive tab back?

latest one but not sure how many days will work.

**Comment by Gregman-js** [comment]
I can confirm that on my S23+ after February update and Nearby Share removal - Quick Share can send files to NearDrop. So probably Quick Share use same protocol for non Samsung devices under the hood as Nearby Share. 
Question is now how we can receive files on phone via Quick Share

**Comment by eugenekv** [comment]
After the update on my s22 Nearby Share also disappeared and when I'm trying to share something with Quick Share the app doesn't see my Mac despite turning on Bluetooth and connection to the same wifi network. Same situation, but it works with Flip 3 but it is searching for the device for a really long time

**Comment by harshbharadwaj07** [comment]
Got a temporary workaround which involves uninstalling the updates for Google Play Services and Quick Share from phone's settings. This works for me atleast for now.

**Comment by dohun0310** [comment]
I change my phone Galaxy Z Filp3 to Pixel 6 Pro. Everything is work. My Pixel 6 Pro still alive `com.google.android.gms/.nearby.sharing.ReceiveSurfaceActivity` activity.

**Comment by eugenekv** [comment]
It's working again after yesterday's update of apps in GP

> After the update on my s22 Nearby Share also disappeared and when I'm trying to share something with Quick Share the app doesn't see my Mac despite turning on Bluetooth and connection to the same wifi network. Same situation, but it works with Flip 3 but it is searching for the device for a really long time

**Comment by NayamAmarshe** [comment]
It seems like they brought back nearby share in the latest files update.

**Comment by grishka** [comment]
Sending works just fine for me. The UI is entirely different but the protocol appears to have remained the same, or at least backwards compatible.
![Screenshot_20240219-135634](https://github.com/grishka/NearDrop/assets/1478258/9d8d2ee7-22a6-4955-b31e-04f2768f54b2)

**Comment by TropicolX** [comment]
> Got a temporary workaround which involves uninstalling the updates for Google Play Services and Quick Share from phone's settings. This works for me atleast for now.

This also works for me on my S21 Ultra

**Comment by grishka** [comment]
As I already said before, the problem with macOS -> Android is that to become discoverable, Android expects BLE advertisements that are impossible to send using any macOS APIs I tried, both public and private. The "service data" field in particular can't be specified, but Android requires it to be present and its value to match a hardcoded prefix. If someone smarter than me would find a way to do it, I'll be more than happy to add it to the app. Or, if someone finds an alternative I may have overlooked while looking through a decompiled Google Services apk.

That "open Google Files" thing is just a gross workaround for being unable to send that BLE advertisement.

**Comment by osenvosem** [comment]
Samsung s23 Ultra. Nearby Share is removed for now and Quick share doesn't see Macbook.

**Comment by makerjeff** [comment]
I have a Samsung Note 10+, and my 2020 M1 Macbook Air doesn't show up under Quick Share's "Share to devices nearby" section. 

Nearby Share + NearDrop was working flawlessly up until yesterday. How would I go about uninstalling Quick Share from the updates as a temporary workaround?

**Comment by harshbharadwaj07** [comment]
> Got a temporary workaround which involves uninstalling the updates for Google Play Services and Quick Share from phone's settings. This works for me atleast for now.

@makerjeff check it once

**Comment by grishka** [comment]
> Wonder if any thoughts about renaming NearDrop to QuickDrop to match the brand change.

No, I'm not Google. I don't change my mind on things 5 times a year for no good reason.

**Comment by harpality** [comment]
> > Wonder if any thoughts about renaming NearDrop to QuickDrop to match the brand change.
> 
> No, I'm not Google. I don't change my mind on things 5 times a year for no good reason.

lol fair

**Comment by AndreiRegiani** [comment]
Doesn't work anymore for me: `Galaxy S23 Ultra, February 1 2024` -> `macOS M1 Pro` using `Quick Share` since NearbyShare is gone.

Mac is found but transfer fails.

**Comment by grishka** [comment]
Most problems seem to be with Samsung devices but I don't have a Samsung device that still receives updates to test this ¯\\\_(ツ)_/¯

Still works with my Pixel 4a.

**Comment by dohun0310** [comment]
@AndreiRegiani 

If it doesn't work on Samsung devices, check the `Share Modal` and make sure there is a `Nearby Share` button. If it does, try to tap the `Nearby Share` button instead of 'Quick Share' - if it doesn't work, I don't know.

**Comment by AndreiRegiani** [comment]
There's no Nearby Share, it was completely removed in the latest update, no
way to enable either in Settings. It's gone.

**Comment by NivlaW** [comment]
> @grishka I think at the moment the best solution is to keep the old Google Files app apk, installation takes a few seconds and then open and use It.
> 
> another solution is we can just clean the storage and cache whenever mac to sends Android file share.
> 
>  screen-20240209-163954.1.1.mp4 
>  Screen.Recording.2024-02-09.at.4.39.03.PM.mov 
> it is not appropriate for the long term but is Ok for now.

THANKKK THIS IS WOK FOR ME

**Comment by nickgieschen** [comment]
Pixel 8. Just upgraded to Sonoma and got the most recent NearDrop (I had the pre-Sonoma one installed). It worked the first time I used it, but after that didn't detect my mac. For what it's worth, the first time I tried (the time it worked), I think my mac was broadcasting both the pre-Sonoma and Sonoma versions of NearDrop, since I saw two devices show up in QuickShare.
